### PR TITLE
Use `AttributeParameters` to get metadata

### PR DESCRIPTION
**Pull request by jvff**
## Motivation

Only the WIT namespace was needed by the `#[wit_import]` and `#[wit_export]` macros, so there was a `util::extract_namespace` helper function to obtain it from the procedural macro's attribute parameters. This helper function used an `AttributeParameters` helper type internally.

However, in order to generate WIT snippets, access to parts of the namespace (specified through the attribute parameters) will be needed, so it makes more sense to skip the helper function and allow access to the `AttributeParameters` helper type directly.

## Proposal

Refactor to move the `extract_namespace` function into an `AttributeParameters::namespace` method, and allow it to be accessed directly by the code generators for the `wit_import` and `wit_export` macros.

## Test Plan

Internal refactoring, no behavior changed so existing tests should catch any regressions.

## Release Plan

Nothing needed.

## Links

- [reviewer checklist](https://github.com/linera-io/linera-protocol/blob/main/CONTRIBUTING.md#reviewer-checklist)

### CODE REVIEW

File: linera-witty-macros/src/util/mod.rs [hunk 1]
```diff
@@ -71,8 +39,19 @@ impl Parse for AttributeParameters {
 }
 
 impl AttributeParameters {
+    /// Parses the attribute parameters to the attribute procedural macro.
+    pub fn new(attribute_parameters: proc_macro::TokenStream) -> Self {
+        syn::parse(attribute_parameters.clone()).unwrap_or_else(|_| {
+            abort!(
+                TokenStream::from(attribute_parameters),
+                r#"Failed to parse attribute parameters, expected either `root = true` \
+                or `package = "namespace:package""#
```
Comment (Twey) [inline linera-witty-macros/src/util/mod.rs:48]: Have we considered using [`darling`](https://docs.rs/darling/latest/darling/)?  I think it should give good error messages by default — without us having to manually duplicate the syntax definition between parser/error.

Comment (jvff) [reply linera-witty-macros/src/util/mod.rs:48]: Interesting, I didn't know about that crate. I took an initial look, and I think I need to experiment with it a bit first to see what the outcome looks like. I'd rather do this some time later :sweat_smile:

Comment (Twey) [reply linera-witty-macros/src/util/mod.rs:48]: If it's a pain — maybe we should make an issue (maybe it's a `good-first-issue`?) and leave it for now :)

**Review by Twey — APPROVED** [review]

File: linera-witty-macros/src/util/mod.rs (hunk 1 above)
Comment (Twey) [reply linera-witty-macros/src/util/mod.rs:48]: I've created this issue (https://github.com/linera-io/linera-protocol/issues/1762) and approved :)

**Pull request merged**